### ISSUE TITLE
Adding a simple TextView to a PhotoViewHolder

### ISSUE THREAD

**Issue by Pantkowsky** [500px/greedo-layout-for-android#28]
Hey guys!

I have a problem with customizing the appearance of items put into greedoLayout. I want to display pictures with a text underneath them. However, if I create a custom layout (relative, linear, whatsoever), the pictures will come out distorted and the app will be very slow and flickering. 

public PhotoViewHolder(ImageView imageView) {
            super(imageView);
            mImageView = imageView;
        }

How can I add a TextView to appear under each of the pictures?

I would highly appreciate if you helped me out!

**Comment by Pantkowsky** [comment]
I've found out that apparently calculateAspectRatio does its calculations of the full view, instead of just a picture, even though this method is called before inflating the layout. Any ideas why?

**Comment by mck182** [comment]
Hey,

thanks for getting in touch. Let me try something and I'll get back to you.

Cheers,
Martin

**Comment by mck182** [comment]
Hey,

so in theory, in your adapter, if you implement the 
```
public double aspectRatioForIndex(int index) { ... }
```
in such way that you will actually return the ratio of the layout itself rather than just the image, the "cell" should be high enough to fit your text.

You could get the image width and height and then add the text height to the image's height (if it's one line and set to fixed size, that's easy, just hardcode the value) and return the aspect ratio calculated from that rather than actually measuring the Layout view.

Hope that helps.

Cheers,
Martin

**Comment by Pantkowsky** [comment]
Hey Martin,

Thanks for the reply! However, I am still struggling with that. The problem is, that the aspect ratios are calculated before the View is even created. For that reason I cannot get the view.height as I will get the null pointer exception. The cycle looks like this (based on the library):
- create adapter
- calculateImageAspectRatio()
- aspectRatioForIndex()
- getLoopedIndex()
- onCreateViewHolder (this is where I need to inflate the view for each item >> image view + text view)
- create holder
- bind

**Comment by Pantkowsky** [comment]
I had actually gotten it to work!

For all the upcoming visitors that might have the same problem:
- in other issues it was pointed out to change some of the internal methods to static in order to work on views rather than imageViews - this did not work for me
- https://github.com/500px/greedo-layout-for-android/issues/10 >> this issue was very helpful for me, especially those lines:

    Size imageViewSize = mSizeCalculator.sizeForChildAtPosition(position);
    holder.itemView.getLayoutParams().width = imageViewSize.getWidth();
    holder.itemView.getLayoutParams().height = imageViewSize.getHeight();

I've added them to onBindViewHolder in the adapter and making GreedoLayoutSizeCalculator a member of adapter class. 

- lastly, I had changed the spacing value:

        int spacing = MeasUtils.dpToPx(0, getContext());
        mRecyclerView.addItemDecoration(new GreedoSpacingItemDecoration(spacing));
  
  to zero. Specifying spacing at this moment cause my item layout (inflated in the adapter) to be cut, resulting in text always being cut in half. If I however specified the padding in xml instead of spacing for recyclerView item decoration, it all worked fine.
